pkg/api/kptfile/v1/types.go

```diff
@@ -276,6 +276,24 @@ type Function struct {
 
 	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
 	ConfigMap map[string]string `yaml:"configMap,omitempty"`
+
+	// `Selectors` are used to specify resources on which the function should be executed
+	// if not specified, all resources are selected
+	Selectors []Selector `yaml:"selectors,omitempty"`
+}
+
+// Selector specifies the selection criteria
+type Selector struct {
+	// APIVersion of the target resources
+	APIVersion string `yaml:"apiVersion,omitempty"`
+	// Kind of the target resources
+	Kind string `yaml:"kind,omitempty"`
+	// Name of the target resources
+	Name string `yaml:"name,omitempty"`
+	// Namespace of the target resources
+	Namespace string `yaml:"namespace,omitempty"`
+	// PackagePath of the target resources
+	PackagePath string `yaml:"packagePath,omitempty"`
```

Comment:
We want to work at `packageName` level.

Reply:
There could be duplicate package names. PackagePath is the relative path of package from the root package directory.